### DISCUSSION TITLE
Target class [livewire] does not exist.

### DISCUSSION THREAD

**Discussion by ousid** [livewire/livewire#4705] [1. Help]
I have been facing this issue for 2 days now, and I cannot find a proper answer.

I'm building a package using `spatie/package-skeleton-laravel` template package, and I did the following.

## Versions:
- __PHP__: "^8.0"
- __Livewire__: "^2.10"

### Dev dependencies
- __Testbench__: "^7.0"
- __Larastan__: "^2.0.1"

## Steps To Reduce
- Clone the repo
- run the `configuration.php` file
- install `livewire`
  - `composer require livewire/livewire`

## Code Example:

From `PackageServiceProvider`

```php
// ..
    public function configurePackage(Package $package): void
    {
        $package
            ->name('laravel-comments')
            ->hasConfigFile()
            ->hasViews('comments')
            ->hasMigration('create_laravel_comments_table');
    }

    public function bootingPackage()
    {
        if (! $this->app->runningInConsole()) {

            $this->registerLivewireComponents();
            
        }
    }

    public function registerLivewireComponents()
    {
        Livewire::component('component-name', ComponentName::class);
    }
// ..
```
- Run Static analysis Test
```bash
./vendor/bin/phpstan
```

## Error Stack Trace:
```
Illuminate\Contracts\Container\BindingResolutionException 

Target class [livewire] does not exist.
```

I'm sure that all my configuration are correct, but I think I'm missing something!

Any help would be much appreciated.

Thanks.

**Comment by joshhanley** [comment]
@ousid thats strange! I've never had an issue with Livewire in my packages. Have you tried doing a `composer dump-autoload`?

**Reply by ousid** [reply]
Hey, @joshhanley. Thanks for the reply.

Yes, I tried it with all possible command such as `composer clearchche`, even tried to re-generate new package, and nothing happens.

**Reply by joshhanley** [reply]
@ousid hmm interesting! Can you push up a test package that demonstrates the issue and so we can pull it and see if we can work out what is going on?

**Reply by ousid** [reply]
Sure thing, @joshhanley

I'll ping you once I push the package.

Thanks.

**Reply by joshhanley** [reply]
@ousid ok thanks!

**Reply by ousid** [reply]
Hey, @joshhanley 

Sorry for the late response.

Here is the repo: https://github.com/ousid/comments

Update:
this is the GitHub actions results of running tests:
https://github.com/ousid/comments/runs/5422151579?check_suite_focus=true

**Reply by ousid** [reply]
Hey, @joshhanley 

I just forgot to mention the solution, the piece I'm missing that, we need to set up the livewire components inside the `TestCase` file under `setUp` method like this

```php
    protected function setUp(): void
    {
        parent::setUp();

      // ...

        $this->registerLivewireComponents();
    }

    protected function registerLivewireComponents()
    {
        Livewire::component('comment', Comment::class);
    }
```

Just got the idea from [Livewire Wizard Package](https://github.com/spatie/laravel-livewire-wizard/blob/main/tests/TestSupport/TestCase.php#L46) By @spatie
